Haskell/Blarney/Backend/Simulation.hs

```diff
@@ -119,10 +118,17 @@ compileSim allSims originalNl = do
            return (nm, outs)
       | o@Net{ netPrim = Output _ nm } <- IArray.elems nl ]
     -- wrap compilation result into a SimulatorIfc
-    return SimulatorIfc { simEffect    = zipWith (>>) effectStreams
-                                                      childrenEffects
-                        , simTerminate = map or (transpose endStreams)
-                        , simOutputs   = Map.fromList outputStreams }
+    let endStream = takeWhileInclusive not (map or $ transpose endStreams)
+    let effectStream = map snd $
+                           zip endStream
+                               (zipWith (>>) effectStreams childrenEffects)
```

Comment:
```
let numCycles = 1 + takeWhile not (map or $ transpose endStreams)
let effectStream = take numCycles (zipWith (>>) effectStreams childrenEffects)
```

?

Reply:
Hmm, my suggestion is a bad idea, it would make things stricter than needed.

Reply:
Still, `takeWhileInclusive` seems unnecessary.  Could do

```
let endStream = False : takeWhile not (map or $ transpose endStreams)
```

Reply:
well... we can't start the `endStream` with `False` and delay the signal... We have to start it with the first element of  `map or $ transpose endStreams`, no?

Reply:
I think it is equivalent to write

```
let endStream = takeWhile not (map or $ transpose endStreams) ++ [True]
```

But then there is the possibility of changing the type of `endStream` from `[Bool]` to `[()]`, because it's the length of the list that matters, not really the values in it.  We could do:

```
let endStream = () : map (const ()) (takeWhile not (map or $ transpose endStreams))
```

Reply:
Just wondering, why did you move the `endStream` stuff into `compileSim`?  All the other signals returned by `compileSim` are infinite lists, so why is this `endStream` special?

Reply:
Well what I took from our previous discussion was that all signals were now truncated to be the same length as `endStream ` upon compilation so that the simulator itself would return infinite lists only if `endStream` is never true, and 
be correctly sized otherwise...

And the `runSim` function now no longer cares and all it does is play the effects and that's it _(and possibly interposes some `IO ()` action between each cycle that could be received as an argument in between each cycle, maybe to give an easy handle for user debug things? Maybe that could be a `SignalMap -> IO ()` that we'd apply to the current slice of the outputs so you can display smthg useful? And if the `endStream` remains `Bool` then we can also use it in there...)_.

Reply:
But this PR doesn't truncate output signals, only `endStream`.

Because `compileSim` is recursive, I would be inclined to do post processing outside that function.  Inside can work too, but removes flexibility.

Reply:
> But this PR doesn't truncate output signals, only `endStream`.

Well I use `map snd $ zip endStream blahblah`

Reply:
Sorry, I missed that!